src/zip_info.rs

```diff
@@ -1,5 +1,59 @@
 pub trait WriteZipInfo {
-    fn write_zip_info(&mut self, exclude: &str) -> String;
+    fn write_zip_info(&mut self, exclude: &str, stat_args: &StatArgs) -> String;
+}
+
+// Struct for our command-line arguments
+#[derive(Debug, RustcDecodable)]
+pub struct Args {
+    pub arg_path: Vec<String>,
+    pub flag_json: bool,
+    pub flag_pretty_json: bool,
+    pub flag_exclude: String,
+    pub flag_compression_type: bool,
+    pub flag_original_size: bool,
+    pub flag_compressed_size: bool,
+    pub flag_compression_rate: bool,
+}
+
+// We will need to pass this down in the main thread, to the individual printers,
+// and there is no point in passing down the whole Args struct, in the spirit of encapsulation.
+#[derive(Debug)]
+pub struct StatArgs {
```

Comment:
This feels a little un-DRY. For the sake of encapsulation, it definitely makes sense to pass the stat args around as their own struct, but I'm wondering whether there's a way we can eliminate the fields from `Args` that are later present in `StatArgs`. That way, we can either construct a full set of args using [composition](https://en.wikipedia.org/wiki/Composition_over_inheritance) (which admittedly I haven't tried in Rust), or just see whether you can read into both structs with `Docopt::new` somehow.

(That, or we can just rename `StatArgs` to something that would make it conceptually distinct from command line arguments lol.)